sycl/test-e2e/lit.cfg.py

```diff
@@ -178,6 +178,63 @@
 else:
     config.substitutions.append( ('%cuda_options', '') )
 
+# Check for device features
+device_features_file='device_features.cpp'
+with open(device_features_file, 'w') as fp:
+    fp.write("""
+#include <sycl/sycl.hpp>
+
+int main() {
+  using namespace sycl;
+
+  auto all_devices = device::get_devices();
+  if (all_devices.size() == 0)
```

Comment:
Can we put SYCL code to a separate file? I think it would easier to support.

Probably we can add a separate [sycl tool](https://github.com/intel/llvm/tree/sycl/sycl/tools) for detecting device features.

Reply:
> Probably we can add a separate [sycl tool](https://github.com/intel/llvm/tree/sycl/sycl/tools) for detecting device features.

I don't want to do that because such a tool must not be part of the user installation, yet we need to run these LIT tests with such user-visible layout (internally).

> Can we put SYCL code to a separate file? I think it would easier to support.

Maybe, as long as it is located under `test-e2e` (see above reasoning) and isn't detected as a test by the LIT framework.

Reply:
> I don't want to do that because such a tool must not be part of the user installation, yet we need to run these LIT tests with such user-visible layout (internally).

I don't see the problem. Can't we exclude this tool from user installation? We don't install neither LIT tool nor e2e tests and they also required running the test. Don't you think that maintaining SYCL sources in a separate file is simpler than within a string in a python script?